src/server/snap.rs

```diff
@@ -239,6 +239,8 @@ impl<R: RaftStoreRouter + 'static> Runnable<Task> for Runner<R> {
                 SNAP_TASK_COUNTER.with_label_values(&["write"]).inc();
                 match self.files.entry(token) {
                     Entry::Occupied(mut e) => {
+                        let limiter = self.snap_mgr.get_limiter();
+                        limiter.request(data.len() as i64, 0);
```

Comment:
How about `data.len()` is greater than `GetSingleBurstBytes()`?
The request API's document says `Caller is responsible to make sure bytes <= GetSingleBurstBytes()`.

Reply:
Any update here?

Reply:
// make sure step <= GetSingleBurstBytes
let total = data.len();
let offset = 0;
loop {
  let left = total - offset;
  if left > step {
    limiter.request(step as i64, 0);
    write(data[offset..offset+step]);
    offset += step;
  } else {
    limiter.request(left, 0);
    write(data[offset..]);
    break;
  }
}

Reply:
I am working on it.